### PR TITLE
Drop Newtonsoft dependency in `LanguageServer.Common`

### PR DESCRIPTION
**Pull request by DoctorKrolic**
1) Moved parsing of well-known elements to built-in `System.Text.Json`
2) We don't need to store `FileInfo`s of each help file since we only use them to read information once
3) Little modernizations, e.g. use `init` keyword since help models are never mutated
4) This allows to drop explicit dependency on `Newtonsoft.Json` from `LanguageServer.Common`. Unfortunatelly, we still have an implicit dependency on this library due to the fact, that OmniSharp depends on it

### CODE REVIEW

File: help/elements.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
     "*.Condition": {
         "description": "Optional expression evaluated to determine whether the target element should be evaluated",
-        "help": "https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions"
+        "helpLink": "https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-conditions"
```
Comment (DoctorKrolic) [inline help/elements.json:4]: Make this name consistent with other files. It is called `helpLink` everywhere else

File: src/LanguageServer.Common/Help/ElementHelp.cs [hunk 2]
```diff
@@ -12,30 +8,11 @@ public class ElementHelp
...
-        /// <summary>
-        ///     Load help property help from JSON.
-        /// </summary>
-        /// <param name="json">
-        ///     A <see cref="JsonReader"/> representing the JSON ("PropertyName": { "description": "PropertyDescription" }).
-        /// </param>
-        /// <returns>
-        ///     A sorted dictionary of help items, keyed by property name.
-        /// </returns>
-        public static SortedDictionary<string, ElementHelp> FromJson(JsonReader json)
```
Comment (DoctorKrolic) [inline src/LanguageServer.Common/Help/ElementHelp.cs:33]: Moved parsing directly to the caller. Don't see any value in spreading the logic across multiple places

**Comment by tintoy** [comment]
This looks great!

Just out of curiosity, does `System.Text.JSON` support custom attributes for explicit JSON property names? I'd hate for a future refactor to accidentally introduce breaking schema changes 🙂

**Comment by tintoy** [comment]
Ah, `[JsonPropertyName]`. Might be worth considering at some point 🙂

Anyway, looks good to me. Thanks!

**Pull request merged**